optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverFactory.java

```diff
@@ -269,6 +270,24 @@
         return new DefaultSolverFactory<>(solverConfig, new SolverConfigContext(kieContainer));
     }
 
+    /**
+     * Introduced as part of https://issues.redhat.com/browse/PLANNER-2068 and is for internal and KieServer use.
+     *
+     * @param kieContainer never null
+     * @param solverConfigResource never null, a classpath resource in the {@link KieContainer}
+     *        as defined by {@link ClassLoader#getResource(String)}
+     * @param scanExcludedPackages a set of packages that will be excluded from scanning when the classpath is being
+     *        scanned for classes annotated as PlanningSolution and PlanningEntity.
+     * @return never null
+     * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+     */
+    public static <Solution_> SolverFactory<Solution_> createFromKieContainerXmlResource(
+            KieContainer kieContainer, String solverConfigResource, Set<String> scanExcludedPackages) {
```

Comment:
Good work on code style, unit testing, etc.

Can we move this `Set<String> scanExcludedPackages` into ScanAnnotatedClassesConfig` as `List<String> packageExcludeList` instead?
https://github.com/kiegroup/optaplanner/blob/7.x/optaplanner-core/src/main/java/org/optaplanner/core/config/domain/ScanAnnotatedClassesConfig.java#L49
Motivation: it fits better there next to the include list, I'd argue. All this is public API that - even in 7.x - we must support for a few years.

From kie server, it would mean not creating the SolverFactory directly from the XML, but using SolverConfig.create() first (so if a ScanAnnotatedClassesConfig is present, it's exclude list can be modified) before creating the SolverFactory with SolverFactory.create(solverConfig). Wdyt?

Reply:
Ok, good idea.

Reply:
So finally as disused the following method was added to the SolverFactory

public static <Solution_> SolverFactory<Solution_> createFromKieContainer(KieContainer kieContainer, SolverConfig solverConfig) 

and kie-server will take care of the rest

Thanks!